volume/driver/overlay_linux_test.go

```diff
@@ -0,0 +1,330 @@
+package driver_test
+
+import (
+	"errors"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/concourse/baggageclaim/volume/driver/driverfakes"
+
+	. "github.com/concourse/baggageclaim/volume/driver"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Overlay", func() {
+
+	Describe("Ancestry", func() {
+
+		type Case struct {
+			input    LiveVolume
+			expected []LiveVolume
+		}
+
+		DescribeTable("cases",
+			func(c Case) {
+				Expect(Ancestry(c.input)).To(Equal(c.expected))
+			},
+			Entry("orphan", Case{
+				input: LiveVolume{Path: "vol1"},
+				expected: []LiveVolume{
+					{Path: "vol1"},
+				},
+			}),
+			Entry("w/ parent", Case{
+				input: LiveVolume{
+					Path: "vol1",
+					Parent: &LiveVolume{
+						Path: "vol2",
+					},
+				},
+				expected: []LiveVolume{
+					{Path: "vol2"},
+					{Path: "vol1", Parent: &LiveVolume{Path: "vol2"}},
+				},
+			}),
+			Entry("w/ granparent", Case{
+				input: LiveVolume{
+					Path: "vol1",
+					Parent: &LiveVolume{
+						Path: "vol2",
+						Parent: &LiveVolume{
+							Path: "vol3",
+						},
+					},
+				},
+				expected: []LiveVolume{
+					{Path: "vol3"},
+					{Path: "vol2", Parent: &LiveVolume{Path: "vol3"}},
+					{Path: "vol1", Parent: &LiveVolume{Path: "vol2", Parent: &LiveVolume{Path: "vol3"}}},
+				},
+			}),
+		)
+
+	})
+
+	Describe("RecoverMountTable", func() {
+
+		var (
+			root    string
+			err     error
+			mounter *driverfakes.FakeMounter
+		)
+
+		JustBeforeEach(func() {
+			err = RecoverMountTable(root, mounter)
+		})
+
+		BeforeEach(func() {
+			mounter = new(driverfakes.FakeMounter)
+		})
+
+		Context("not having a live dir", func() {
+			It("does nothing", func() {
+				Expect(err).ToNot(HaveOccurred())
+			})
+		})
+
+		Context("having a live dir", func() {
+
+			BeforeEach(func() {
+				root, err = ioutil.TempDir("", "bclaim_ovl_test")
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			AfterEach(func() {
+				os.RemoveAll(root)
+			})
+
+			Context("having no volumes", func() {
+				It("succeeds", func() {
+					Expect(err).ToNot(HaveOccurred())
+				})
+			})
+
+			Context("having a file in the middle of vol dirs", func() {
+				BeforeEach(func() {
+					_, err = os.Create(filepath.Join(root, "file"))
+					Expect(err).ToNot(HaveOccurred())
+				})
+
+				It("fails", func() {
```

Comment:
Seems a bit brittle to error on this.
WDYT ?

Reply:
My thinking was on the lines of "if there's a file here, something went reeeaally wrong, possibly in the config" - perhaps the user misconfigured the location on where volumes live.

By failing here (in a method that's used only during startup time), the user can get that feedback that something is on the way, rather than later through the silence 🤔